client/src/store/comments/actions.ts

```diff
@@ -45,3 +48,44 @@ export const getComments = (postId: string, page: number): (dispatch: Dispatch)
             dispatch(showAlert(e.response.data.message, 'danger'));
         }
     };
+
+export const editComment = (comment: string, commentId: string)
+    : { type: string, payload: any } => (
+    {
+        type: EDIT_COMMENT,
+        payload: {
+            comment,
+            commentId,
+        },
+    });
+
+export const changeComment = (comment: string, commentId: string)
+    : { type: string, payload: any } => (
+    {
+        type: CHANGE_COMMENT,
+        payload: {
+            comment,
+            commentId,
+        },
+    });
+
+export const changeEditStatus = (commentId: string): { type: string, payload: string } => ({
+    type: CHANGE_EDIT_STATUS_COMMENT,
+    payload: commentId,
+});
+
+export const editCommentAsync = (
+    comment: string,
+    commentId: string,
+    email: string,
+): (dispatch: Dispatch) => Promise<void> =>
+    async (dispatch: Dispatch): Promise<void> => {
+        try {
+            dispatch(editComment(comment, commentId));
```

Comment:
why do you save changes before call to backend?

Reply:
Fixed